File: download/src/main/java/org/sqlunet/download/DownloadActivity.java
/*
 * Copyright (c) 2022. Bernard Bou
 */

package org.sqlunet.download;

import android.os.Bundle;
import android.util.Log;

import androidx.annotation.Nullable;
import androidx.appcompat.app.ActionBar;
import androidx.appcompat.app.AppCompatActivity;
import androidx.appcompat.widget.Toolbar;

import static org.sqlunet.download.BaseDownloadFragment.DOWNLOAD_DOWNLOADER_ARG;

@FunctionalInterface
interface OnComplete
{
	void onComplete(boolean success);
}

/**
 * Download activity
 * This activity does not handle the completion signal.
 *
 * @author <a href="mailto:1313ou@gmail.com">Bernard Bou</a>
 */
public class DownloadActivity extends AppCompatActivity implements OnComplete
{
	static private final String TAG = "DownloadA";

	@Override
	protected void onCreate(@Nullable final Bundle savedInstanceState)
	{
		super.onCreate(savedInstanceState);

		// downloader
		final String overriddenDownloader = getIntent().getStringExtra(DOWNLOAD_DOWNLOADER_ARG);
		final Settings.Downloader downloader = overriddenDownloader == null ? Settings.Downloader.getFromPref(this) : Settings.Downloader.valueOf(overriddenDownloader);

		// content
		setContentView(R.layout.activity_download);

		// toolbar
		final Toolbar toolbar = findViewById(R.id.toolbar);
		setSupportActionBar(toolbar);

		// set up the action bar
		final ActionBar actionBar = getSupportActionBar();
		if (actionBar != null)
		{
			actionBar.setDisplayShowTitleEnabled(true);
		}

		if (savedInstanceState == null)
		{
			// fragment
			BaseDownloadFragment downloadFragment = null;
			switch (downloader)
			{
				case DOWNLOAD_SERVICE:
					downloadFragment = new DownloadFragment();
					break;

				case DOWNLOAD_ZIP_SERVICE:
					downloadFragment = new DownloadZipFragment();
					break;
			}
			// pass arguments over to fragment
			downloadFragment.setArguments(getIntent().getExtras());

			getSupportFragmentManager() //
					.beginTransaction() //
					.setReorderingAllowed(true) //
					.replace(R.id.container, downloadFragment) //
					.commit();
		}
	}

	@Override
	public void onComplete(final boolean success)
	{
		Log.d(TAG, "OnComplete " + success + " " + this);
	}
}

File: common/src/main/java/org/sqlunet/browser/config/DownloadActivity.kt
/*
 * Copyright (c) 2023. Bernard Bou
 */
package org.sqlunet.browser.config

import android.os.Bundle
import android.util.Log
import android.view.Menu
import android.view.MenuItem
import androidx.appcompat.app.ActionBar
import com.bbou.download.workers.DownloadActivity
import org.sqlunet.browser.EntryActivity.Companion.rerun
import org.sqlunet.browser.MenuHandler
import org.sqlunet.browser.common.R

/**
 * Download activity
 * Handles completion by re-entering application through entry activity.
 *
 * @author [Bernard Bou](mailto:1313ou@gmail.com)
 */
class DownloadActivity : DownloadActivity() {

    public override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        // set up the action bar
        val actionBar = supportActionBar
        if (actionBar != null) {
            actionBar.displayOptions = ActionBar.DISPLAY_SHOW_HOME or ActionBar.DISPLAY_HOME_AS_UP or ActionBar.DISPLAY_SHOW_TITLE
        }
    }

    override fun onComplete(success: Boolean) {
        Log.d(TAG, "OnComplete $success $this")
        if (success) {
            rerun(this)
        }
    }

    // M E N U

    override fun onCreateOptionsMenu(menu: Menu): Boolean {
        // inflate the menu; this adds items to the type bar if it is present.
        menuInflater.inflate(R.menu.initialize, menu)
        // MenuCompat.setGroupDividerEnabled(menu, true);
        return true
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        // handle home
        if (item.itemId == android.R.id.home) {
            Log.d(TAG, "onHomePressed")
            rerun(this)
            return true
        }
        return MenuHandler.menuDispatchWhenCantRun(this, item)
    }

    companion object {
        private const val TAG = "DownloadA"
    }
}
